### PR TITLE
feat:Make an Apache Shenyu SSO authentication plugin based on Casdoor #4

### LINKED ISSUE #4: Make an Apache Shenyu SSO authentication plugin based on Casdoor
See:

- oauth2-plugin: https://shenyu.apache.org/docs/next/plugin-center/security/oauth2-plugin
- how to develop a plugin: https://shenyu.apache.org/docs/developer/custom-plugin/

About how to develop a plugin to integrate Casdoor with a gateway software, plz refer to:

- APISIX: https://apisix.apache.org/docs/apisix/plugins/authz-casdoor/
- Kong: https://github.com/casdoor/casdoor/issues/280

Example repo created here: https://github.com/casdoor/shenyu-sso

### PR DESCRIPTION
**Pull request by jakiuncle**
Fix: https://github.com/casdoor/shenyu-plugin-auth/issues/4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,38 @@
-# shenyu-sso
\ No newline at end of file
+# shenyu-sso
```
Comment (hsluoyz) [inline README.md]: shenyu-plugin-auth

File: shenyu-plugin-auth/src/main/java/org/apache/shenyu/plugin/auth/AuthPlugin.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+package org.apache.shenyu.plugin.auth;
```
Comment (hsluoyz) [inline shenyu-plugin-auth/src/main/java/org/apache/shenyu/plugin/auth/AuthPlugin.java:14]: Apache headers for all source files

File: shenyu-spring-boot-starter-plugin-auth/shenyu-spring-boot-starter-pulgin-auth.iml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hsluoyz) [inline shenyu-spring-boot-starter-plugin-auth/shenyu-spring-boot-starter-pulgin-auth.iml]: All .iml files should be ingored: https://stackoverflow.com/questions/14251624/intellij-idea-hide-iml-files

File: README.md [hunk 4]
```diff
@@ -1 +1,38 @@
...
+
+### Stpe2.Add this plugin to ShenYu.
+
+1. Move shenyu-plugin-auth to shenyu-plugin.
+2. Move shenyu-spring-boot-starter-plugin-auth to shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin
+3. Add dependency in shenyu-bootstrap/pom.xml
+
+![dependency](static/img/dependency.png)
+
+### Step3.Withelist Configure
```
Comment (hsluoyz) [inline README.md]: A lot of typos for `withe`

File: shenyu-spring-boot-starter-plugin-auth/pom.xml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>shenyu-spring-boot-starter-plugin</artifactId>
+        <groupId>org.apache.shenyu</groupId>
+        <version>2.5.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>shenyu-spring-boot-starter-pulgin-auth</artifactId>
```
Comment (hsluoyz) [inline shenyu-spring-boot-starter-plugin-auth/pom.xml]: A lot of typos for `pulg`

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jakiuncle) [reply README.md]: ok

File: shenyu-plugin-auth/src/main/java/org/apache/shenyu/plugin/auth/AuthPlugin.java (hunk 2 above)
Comment (jakiuncle) [reply shenyu-plugin-auth/src/main/java/org/apache/shenyu/plugin/auth/AuthPlugin.java:14]: ok

File: shenyu-spring-boot-starter-plugin-auth/shenyu-spring-boot-starter-pulgin-auth.iml (hunk 3 above)
Comment (jakiuncle) [reply shenyu-spring-boot-starter-plugin-auth/shenyu-spring-boot-starter-pulgin-auth.iml]: ok

File: README.md (hunk 4 above)
Comment (jakiuncle) [reply README.md]: ok

File: shenyu-spring-boot-starter-plugin-auth/pom.xml (hunk 5 above)
Comment (jakiuncle) [reply shenyu-spring-boot-starter-plugin-auth/pom.xml]: ok

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**